tensorly/backend/core.py

```diff
@@ -833,8 +833,110 @@ def truncated_svd(self, matrix, n_eigenvecs=None, **kwargs):
         U, S, V = U[:, :n_eigenvecs], S[:n_eigenvecs], V[:n_eigenvecs, :]
         return U, S, V
 
+    def randomized_svd(self, matrix, n_eigenvecs=None, n_oversamples=5, n_iter=2, random_state=None, **kwargs):
+        """Computes a truncated randomized SVD.
+
+        If `n_eigenvecs` is specified, sparse eigendecomposition is used on
+        either matrix.dot(matrix.T) or matrix.T.dot(matrix).
+
+        Parameters
+        ----------
+        matrix : tensor
+            A 2D tensor.
+        n_eigenvecs : int, optional, default is None
+            If specified, number of eigen[vectors-values] to return.
+        n_oversamples: int, optional, default = 5
+            rank overestimation value for finiding subspace with better allignment
+        n_iter: int, optional, default = 2
+            number of power iterations for the `randomized_range_finder` subroutine
+        random_state: {None, int, np.random.RandomState}
+        **kwargs : optional
+            kwargs are used to absorb the difference of parameters among the other SVD functions
+
+        Returns
+        -------
+        U : 2-D tensor, shape (matrix.shape[0], n_eigenvecs)
+            Contains the right singular vectors
+        S : 1-D tensor, shape (n_eigenvecs, )
+            Contains the singular values of `matrix`
+        V : 2-D tensor, shape (n_eigenvecs, matrix.shape[1])
+            Contains the left singular vectors
+
+        Notes
+        -----
+        This function is implemented based on Algorith 5.1 in `Finding structure with randomness:
+        Probabilistic algorithms for constructing approximate matrix decompositions`
+        - Halko et al (2009)
+        """
+
+        # hack to avoid the warning in partial svd
+        def _partial_svd(matrix, n_eigenvecs=n_eigenvecs, random_state=None):
```

Comment:
Why did you use this? This will also only work for the NumPy backend if we use scipy.linalg.svd.